In [48]:
import os
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objects as go

import datapane as dp

In [49]:
dir_path = os.getcwd()

dir_path

'W:\\Github\\Personal Projects\\sales-analysis-reporting\\Notebook'

In [50]:
path_to_files = './../Sales_Data_Monthly'
files = [file for file in os.listdir(path_to_files) if not file.startswith('.')]
len(files)

12

In [51]:
path_to_combined_csv = './../Sales_Data_Yearly'
sales_data_combined = pd.DataFrame()

for file in files:
    current_month_data = pd.read_csv(path_to_files + '/' + file)
    sales_data_combined = pd.concat([sales_data_combined, current_month_data])

sales_data_combined.to_csv(path_to_combined_csv + '/combined_yearly_data_2019.csv')

In [52]:
df = pd.read_csv(path_to_combined_csv + '/combined_yearly_data_2019.csv')
df.head()

Unnamed: 0 Order ID                     Product Quantity Ordered  \
0           0   174083  Bose SoundSport Headphones                1   
1           1   174084    Lightning Charging Cable                1   
2           2   174085                 Macbook Pro                1   
3           3   174085             Normal Keyboard                1   
4           4   174086  Bose SoundSport Headphones                1   

  Price Each      Order Date                        Purchase Address  
0      99.99  04/14/19 20:01            565 2nd St, Dallas, TX 75001  
1      14.95  04/05/19 21:00  628 Spruce St, San Francisco, CA 94016  
2     1699.0  04/08/19 15:27          289 Elm St, Portland, ME 04101  
3      99.99  04/08/19 15:27          289 Elm St, Portland, ME 04101  
4      99.99  04/27/19 13:37          883 West St, Seattle, WA 98101

In [53]:
df.shape

(189801, 7)

In [54]:
df = df.drop(['Unnamed: 0'], axis = 1)
df.shape

(189801, 6)

In [55]:
temp_df = df[df['Order Date'].str[0:2] == 'Or']
temp_df

Order ID  Product  Quantity Ordered  Price Each  Order Date  \
1220    Order ID  Product  Quantity Ordered  Price Each  Order Date   
2189    Order ID  Product  Quantity Ordered  Price Each  Order Date   
2258    Order ID  Product  Quantity Ordered  Price Each  Order Date   
2504    Order ID  Product  Quantity Ordered  Price Each  Order Date   
3845    Order ID  Product  Quantity Ordered  Price Each  Order Date   
...          ...      ...               ...         ...         ...   
187310  Order ID  Product  Quantity Ordered  Price Each  Order Date   
187499  Order ID  Product  Quantity Ordered  Price Each  Order Date   
188133  Order ID  Product  Quantity Ordered  Price Each  Order Date   
188297  Order ID  Product  Quantity Ordered  Price Each  Order Date   
189473  Order ID  Product  Quantity Ordered  Price Each  Order Date   

        Purchase Address  
1220    Purchase Address  
2189    Purchase Address  
2258    Purchase Address  
2504    Purchase Address  
3845    Purchase Address  
...                  ...  
187310  Purchase Address  
187499  Purchase Address  
188133  Purchase Address  
188297  Purchase Address  
189473  Purchase Address  

[346 rows x 6 columns]

366 Order Date columns when there are 360 total item counts
346 when there are 362 items

In [57]:
df = df[df['Order Date'].str[0:2] != 'Or']
df.head()

Order ID                     Product Quantity Ordered Price Each  \
0   174083  Bose SoundSport Headphones                1      99.99   
1   174084    Lightning Charging Cable                1      14.95   
2   174085                 Macbook Pro                1     1699.0   
3   174085             Normal Keyboard                1      99.99   
4   174086  Bose SoundSport Headphones                1      99.99   

       Order Date                        Purchase Address  
0  04/14/19 20:01            565 2nd St, Dallas, TX 75001  
1  04/05/19 21:00  628 Spruce St, San Francisco, CA 94016  
2  04/08/19 15:27          289 Elm St, Portland, ME 04101  
3  04/08/19 15:27          289 Elm St, Portland, ME 04101  
4  04/27/19 13:37          883 West St, Seattle, WA 98101

In [59]:
df.shape

(189455, 6)